digits/test_scheduler.py

```diff
@@ -46,6 +47,9 @@ def test_add_remove_job(self):
         job = Job('tmp')
         assert self.s.add_job(job), 'failed to add job'
         assert len(self.s.jobs) == 1, 'scheduler has %d jobs' % len(self.s.jobs)
-        assert self.s.delete_job(job), 'failed to delete job'
+        try:
+            self.s.delete_job(job)
+        except errors.DeleteError as e:
+            raise AssertionError(e.__str__())
```

Comment:
No need to catch the error and raise an AssertionError - just let the DeleteError go through.